src/main/java/prefs/Preferences.java

```diff
@@ -128,15 +138,32 @@ public void setLastOpenBoard(String board) {
     }
     
     public Optional<String> switchBoard() {
-        if (getLastOpenBoard().isPresent() && getAllBoards().size() > 1) {
-            List<String> boardNames = getAllBoardNames();
+        setBoardToSwitch();
+        return getLastOpenBoard();
+    }
+
+    /**
+     * Sets the lastOpenBoard to the board to switch to
+     */
+    private void setBoardToSwitch(){
```

Comment:
Only the preferences one is unmerged and making progress, so I guess you can go ahead. You can look at the PR to see what the interface might be like. Probably better for the others to adapt to the new design here anyway.

In any case, do try to get feedback early and iterate on the design!